llama_index/indices/prompt_helper.py

```diff
@@ -139,38 +139,31 @@ def _get_available_context_size(self, prompt: BasePromptTemplate) -> int:
             available context window = total context window
                 - input (partially filled prompt)
                 - output (room reserved for response)
+
+        Notes:
+        - Available context size is further clamped to be non-negative.
         """
         empty_prompt_txt = get_empty_prompt_txt(prompt)
-        prompt_tokens = self._tokenizer(empty_prompt_txt)
-        num_prompt_tokens = len(prompt_tokens)
-
-        return self.context_window - num_prompt_tokens - self.num_output
+        num_empty_prompt_tokens = len(self._tokenizer(empty_prompt_txt))
+        return max(self.context_window - num_empty_prompt_tokens - self.num_output, 0)
```

Comment:
A negative infer is incorrect.  Would you like:

- Blow up with an `Exception` if infer negative
- Truncate to 0 (what I did)

Now thinking about it, I think blowing up with an `Exception` may be preferable. What do you think?

Reply:
I decided that truncating to 0 could be a vector for sneaky bugs.  So I made it instead validate zero or positive 👍 🚀